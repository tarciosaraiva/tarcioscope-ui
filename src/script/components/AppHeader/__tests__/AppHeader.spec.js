import { shallow } from '@vue/test-utils'

import AppHeader from '../AppHeader.vue'

describe('<AppHeader />', () => {
  let wrapper

  describe('created behaviour', () => {
    beforeEach(() => {
      jest.spyOn(AppHeader.methods, 'defineLogoHeight').mockImplementation(() => {})
      wrapper = shallow(AppHeader)
    })

    afterEach(() => {
      jest.restoreAllMocks()
    })

    it('renders a footer block', () => {
      expect(wrapper.contains('header')).toBe(true)
    })

    it('matches snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })

    it('should have invoked "defineLogoHeight"', () => {
      expect(AppHeader.methods.defineLogoHeight).toHaveBeenCalled()
    })

    it('should have data with height property and a default value', () => {
      expect(wrapper.vm.height).toEqual(30)
    })
  })

  describe('#defineLogoHeight', () => {
    beforeEach(() => {
      wrapper = shallow(AppHeader)
    })

    afterEach(() => {
      jest.restoreAllMocks()
    })

    it('should have kept height at 30 when window width is lower than 768 px', () => {
      jest.spyOn(wrapper.vm, 'windowWidthGreaterThan768').mockImplementation(() => false)
      wrapper.vm.defineLogoHeight()
      expect(wrapper.vm.height).toEqual(30)
    })

    it('should have kept height at 60 when window width is greater than 768 px', () => {
      jest.spyOn(wrapper.vm, 'windowWidthGreaterThan768').mockImplementation(() => true)
      wrapper.vm.defineLogoHeight()
      expect(wrapper.vm.height).toEqual(60)
    })
  })
})
