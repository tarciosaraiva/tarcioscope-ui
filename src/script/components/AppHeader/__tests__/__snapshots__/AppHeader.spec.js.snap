// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppHeader /> created behaviour matches snapshot 1`] = `
<header class="app-header">
  <img height="30" src="../../../img/telescope.svg" alt="tarcioscope">
  <h1>tarcioscope</h1>
</header>
`;
